nativescript/src/tns-java-classes.ts

```diff
@@ -0,0 +1,25 @@
+//Resolve JavaScript classes that extend a Java class, and need to resolve
+//their JavaScript module from a bundled script. For example:
+//NativeScriptApplication, NativeScriptActivity, etc.
+//
+//This module gets bundled together with the rest of the app code and the
+//`require` calls get resolved to the correct bundling import call.
+//
+//At runtime the module gets loaded *before* the rest of the app code, so code
+//placed here needs to be careful about its dependencies.
```

Comment:
In newer versions the same functionality is covered by vendor.ts and vendor-platform.android/ios.ts, they look just as bad, but they're from a template so I don't think we should alter them too much.